lib/http-proxy.js

```diff
@@ -1,7 +1,7 @@
 var http      = require('http'),
     https     = require('https'),
     url       = require('url'),
-    httpProxy = require('./http-proxy/');
+    httpProxy = require('./http-proxy/index.js');
```

Comment:
Could we just add a comment on why this has to happen? Otherwise it may be removed in the future without context